doc/man3/EVP_EncryptInit.pod

```diff
@@ -84,18 +87,27 @@ EVP_CIPHER_do_all_provided
 
  int EVP_EncryptInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
                         ENGINE *impl, const unsigned char *key, const unsigned char *iv);
+ int EVP_EncryptInit_ex2(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
+                         const unsigned char *key, const unsigned char *iv,
+                         const OSSL_PARAM params[]);
  int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out,
                        int *outl, const unsigned char *in, int inl);
  int EVP_EncryptFinal_ex(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl);
 
  int EVP_DecryptInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
                         ENGINE *impl, const unsigned char *key, const unsigned char *iv);
+ int EVP_DecryptInit_ex2(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
+                         const unsigned char *key, const unsigned char *iv,
+                         const OSSL_PARAM params[]);
  int EVP_DecryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out,
                        int *outl, const unsigned char *in, int inl);
  int EVP_DecryptFinal_ex(EVP_CIPHER_CTX *ctx, unsigned char *outm, int *outl);
 
  int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
                        ENGINE *impl, const unsigned char *key, const unsigned char *iv, int enc);
+ int EVP_CipherInit_ex2(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
+                        const unsigned char *key, const unsigned char *iv,
+                        int enc, const OSSL_PARAM params[]);
```

Comment:
Should `EVP_EncryptInit_ex`,  `EVP_DecryptInit_ex` and `EVP_CipherInit_ex` become deprecated?

Reply:
Note: I have zero concerns if this gets answered by a subsequent PR

Reply:
That's a question for the OTC, not here.

Reply:
I suspect `EVP_CipherInit_ex()` should be.  Why not `EVP_CipherInit()` too?

Reply:
The other question is should we follow the new naming convention.  I.e. change `EVP_CipherInit_ex2()` to `EVP_CIPHER_init()`, `EVP_CIPHER_init_ex()` or even `EVP_CIPHER_init_ex2()`.

Reply:
> The other question is should we follow the new naming convention.  I.e. change `EVP_CipherInit_ex2()` to `EVP_CIPHER_init()`, `EVP_CIPHER_init_ex()` or even `EVP_CIPHER_init_ex2()`.

It should be the *whole* set of EVP_Cipher functions, not just the init.  I believe we have a vote already, more or less saying "not now"

Reply:
Not now for renaming old functions.  We didn't discuss new.
E.g. we could deprecate `EVP_CIPHER_init()` and `EVP_CIPHER_init_ex()` and name the new function `EVP_CIPHER_init()`.

Reply:
IMO the not now was also for renaming the new functions if they would not fit into the whole set. It would be super ugly to have EVP_CIPHER_init() but then have to call EVP_CipherUpdate() and EVP_CipherFinal().

Reply:
Adding aliases for EVP_CipherUpdate and EVP_CipherFinal isn't onerous :)

Reply:
I've created #14492 to deal with the deprecation of the _ex calls that take an ENGINE * argument.